systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -168,24 +177,87 @@ private static void verifyCerts(String certificate, String component) {
+    @Tag(INTERNAL_CLIENTS_USED)
+    @Tag(ROLLING_UPDATE)
+    void testAutoRenewClusterCaCertsTriggeredByAnno(ExtensionContext extensionContext) {
+        autoRenewSomeCaCertsTriggeredByAnno(
+                extensionContext,
+                /* ZK node need new certs */
+                true,
+                /* brokers need new certs */
+                true,
+                /* eo needs new cert */
+                true);
+    }
+
+    @IsolatedTest("Using more tha one Kafka cluster in one namespace")
+    @Tag(INTERNAL_CLIENTS_USED)
+    @Tag(ROLLING_UPDATE)
+    void testAutoRenewClientsCaCertsTriggeredByAnno(ExtensionContext extensionContext) {
+        autoRenewSomeCaCertsTriggeredByAnno(
+            extensionContext,
+                /* no communication between clients and zk, so no need to roll */
+                false,
+                /* brokers need to trust client certs with new cert */
+                true,
+                /* eo needs to generate new client certs */
+                true);
+    }
+
+    @IsolatedTest("Using more tha one Kafka cluster in one namespace")
+    @Tag(ACCEPTANCE)
+    @Tag(INTERNAL_CLIENTS_USED)
+    @Tag(ROLLING_UPDATE)
+    void testAutoRenewAllCaCertsTriggeredByAnno(ExtensionContext extensionContext) {
+        autoRenewSomeCaCertsTriggeredByAnno(
+            extensionContext,
+                true,
+                true,
+                true);
+    }
+
+    @SuppressWarnings({"checkstyle:MethodLength", "checkstyle:NPathComplexity"})
     void autoRenewSomeCaCertsTriggeredByAnno(
-            final List<String> secretsToAnnotate,
+            ExtensionContext extensionContext,
             boolean zkShouldRoll,
             boolean kafkaShouldRoll,
             boolean eoShouldRoll) {
 
-        createKafkaCluster();
+        createKafkaCluster(extensionContext);
+
+        String kafkaClientsName = mapTestWithKafkaClientNames.get(extensionContext.getDisplayName());
+        String clusterName = mapTestWithClusterNames.get(extensionContext.getDisplayName());
+        String topicName = mapTestWithTestTopics.get(extensionContext.getDisplayName());
+        String userName = mapTestWithTestUsers.get(extensionContext.getDisplayName());
+        List<String> secrets = null;
+
+        // to make it parallel we need decision maker...
+        switch (extensionContext.getDisplayName()) {
+            case "testAutoRenewClusterCaCertsTriggeredByAnno":
+                secrets = Arrays.asList(clusterCaCertificateSecretName(clusterName));
+                break;
+            case "testAutoRenewClientsCaCertsTriggeredByAnno":
+                secrets = Arrays.asList(clientsCaCertificateSecretName(clusterName));
+                break;
+            case "testAutoRenewAllCaCertsTriggeredByAnno":
+                secrets = Arrays.asList(clusterCaCertificateSecretName(clusterName),
+                    clientsCaCertificateSecretName(clusterName));
+                break;
```

Comment:
I think it's not a good way to use the test name in the switch. Would be better some annotation/tag ?